File: php/src/Apps/AppsService.php
<?php

declare(strict_types=1);

namespace Buble\Apps;

use Buble\Http\TransportInterface;

final class AppsService
{
    private AppGenerationsService $generations;

    public function __construct(private readonly TransportInterface $transport)
    {
        $this->generations = new AppGenerationsService($transport);
    }

    /**
     * @return array<string, mixed>
     */
    public function list(): array
    {
        return $this->transport->request('GET', '/api/v1/apps');
    }

    /**
     * @return array<string, mixed>
     */
    public function retrieve(string $appId): array
    {
        return $this->transport->request('GET', '/api/v1/apps/' . rawurlencode($appId));
    }

    public function generations(): AppGenerationsService
    {
        return $this->generations;
    }
}

File: dotnet/src/Buble.Sdk/Apps/AppsService.cs
using Buble.Sdk.Http;

namespace Buble.Sdk.Apps;

public sealed class AppsService
{
    private readonly BubleHttpClient _http;

    internal AppsService(BubleHttpClient http)
    {
        _http = http;
        Generations = new AppGenerationsService(http);
    }

    public AppGenerationsService Generations { get; }

    public Task<Envelope<IReadOnlyList<PublicApp>>?> ListAsync(CancellationToken cancellationToken = default)
    {
        return _http.GetAsync<Envelope<IReadOnlyList<PublicApp>>>("/api/v1/apps", cancellationToken: cancellationToken);
    }

    public Task<Envelope<PublicApp>?> RetrieveAsync(string appId, CancellationToken cancellationToken = default)
    {
        return _http.GetAsync<Envelope<PublicApp>>(
            "/api/v1/apps/" + BubleHttpClient.EncodePathSegment(appId),
            cancellationToken: cancellationToken);
    }
}
